srml/support/src/storage/mod.rs

```diff
@@ -226,6 +226,12 @@ pub trait StorageDoubleMap<K1: FullEncode, K2: FullEncode, V: FullCodec> {
 		KArg1: EncodeLike<K1>,
 		KArg2: EncodeLike<K2>;
 
+	/// Swap the values of two key-pairs.
+	fn swap<KArg1, KArg2>(key11: KArg1, key12: KArg2, key21: KArg1, key22: KArg2)
```

Comment:
I would like to see 4 generic parameters here. Otherwise we force the user to have 2 times the same type. That is not something we wanted to achieve with `EncodeLike`.